app/components/session.service.js

```diff
@@ -0,0 +1,58 @@
+;(function () {
+  angular
+    .module("todo")
+    .service("SessionService", SessionService);
+
+  SessionService.$inject = ["$http", "API", "$state"];
+
+  function SessionService($http, API, $state) {
+    var self = this;
+    self.user = {
+      userView: "list"
+    };
+    self.getUser = getUser;
+    self.login = login;
+    self.registration = registration;
+    self.logout = logout;
+
+    self.getUser();
+
+    function getUser() {
+      $http.get(API.BASE + API.AUTH + API.INFO)
+        .then(function (response) {
+          self.user.data = response.data.user;
+          $state.go("groups.list");
```

Comment:
what will happen if I press `F5` and refresh browser page? I expect to be on the same page. Could you please check it?